flytekit/deck/deck.py

```diff
@@ -174,3 +174,72 @@ def get_deck_template() -> "Template":
         autoescape=select_autoescape(enabled_extensions=("html",)),
     )
     return env.get_template("template.html")
+
+
+class PythonDependencyDeck(Deck):
+    """
+    PythonDependencyDeck is a deck that contains information about the library name and version.
+    """
+
+    def __init__(self, name: Optional[str] = "Python Dependency", html: Optional[str] = ""):
+        super().__init__(name, html)
+
+    @property
+    def html(self) -> str:
+        import json
+        import subprocess
+        import sys
+
+        import pandas as pd
+        from flytekitplugins.deck.renderer import TableRenderer
+
+        try:
+            installed_packages = json.loads(
+                subprocess.check_output([sys.executable, "-m", "pip", "list", "--format", "json"])
+            )
+            requirements_txt = ""
+
+            for package_info in installed_packages:
+                package_name = package_info["name"]
+                package_version = package_info["version"]
+                requirements_txt += f"{package_name}=={package_version}\n"
```

Comment:
what if we just run `pip freeze` instead?

```suggestion
            requirements_txt = subprocess.check_output(["pip", "freeze"]).decode("utf-8").replace("\\n", "\n")
```

Reply:
updated